File: 5_control_flow/control_flow.go
package main

import "fmt"

/* Define una función que encuentre la media aritmetica de los numeros reales en el slice */

func media(x []float64) (m float64) {
    total := 0.0
    if len(x) == 0 {
        m = 0
    } else {
        for _,v := range x {
            total += v
        }
        m = total / float64(len(x))
    }
    return
}

func main() {
    x := []float64{2.15, 3.14, 42.0, 29.5}
    fmt.Println(media(x))       //19.197499999999998
}

File: 5_control_flow/control_flow.c
#include <stdio.h>

double media(double x[], int length) {
    double total = 0.0;
    double m;
    if (length == 0) {
       m = 0;
    } else {
       for (int i = 0; i < length; i++){
          total += x[i];
       }
       m = total / (double) length;
    }
    return m;
}


int main() {
    double x[] = {2.15, 3.14, 42.0, 29.5};
    int length = sizeof(x)/sizeof(x[0]);
    printf("%lf \n", media(x, length));           //19.127499999999998

    return 0;
}

    
